plutus-core/plutus-ir/src/PlutusIR/Compiler/Recursion.hs

```diff
@@ -98,15 +98,16 @@ mkFixpoint bs = do
           name <- liftQuote $ toProgramName fixByKey
           let (fixByTerm, fixByType) = Function.fixByAndType
           pure (PLC.Def (PLC.VarDecl noProvenance name (noProvenance <$ fixByType)) (noProvenance <$ fixByTerm, Strict), mempty)
-    fixBy <- lookupOrDefineTerm p0 fixByKey mkFixByDef
```

Comment:
This change does make sense to me: the `fixBy` is generated only if arity /= 0.

Reply:
> `if arity /= 0`

If `arity /= 1`, because `0` isn't supported at all (we're in a strict language) and `1` is handled by the Z combinator.

Reply:
you're right, I made a typo.